src/t_stream.c

```diff
@@ -730,6 +726,9 @@ int64_t streamTrim(stream *s, streamAddTrimArgs *args) {
         }
 
         if (remove_node) {
+            /* Check if we exceeded the amount of work we could do */
+            if (limit && (deleted + entries) > limit)
+                break;
```

Comment:
maybe i'm missing something, but do you check this condition inside `if (remove_node`? can't we check it just after `int64_t entries = lpGetInteger(p);`?

Reply:
My head's not spinning. I'm wrong.